consensus/src/chained_bft/liveness/proposal_generator.rs

```diff
@@ -199,19 +200,26 @@ impl<T: Payload> ProposalGenerator<T> {
             }
         };
 
-        let block_store = Arc::clone(&self.block_store);
-        match self
-            .txn_manager
-            .pull_txns(self.max_block_size, exclude_payload)
-            .await
+        // Reconfiguration rule - we propose empty blocks after reconfiguration until it's committed
+        let txns = if hqc_block
+            .compute_result()
+            .executed_state
+            .validators
```

Comment:
nit: may be add a wrapper function `is_reconfiguration()` to the compute result?